From 7db17d32cef1ab044de5ece298cf7786fc5d2a6f Mon Sep 17 00:00:00 2001
From: "Rajat Soni(Temp)" <quic_rajson@quicinc.com>
Date: Mon, 11 Jul 2022 17:10:54 +0530
Subject: [PATCH] ath12k: fill pdev id for fw test cmd

Fill the pdev id for the fw test cmd in driver.

Signed-off-by: Rajat Soni(Temp) <quic_rajson@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/testmode.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/testmode.c b/drivers/net/wireless/ath/ath12k/testmode.c
index 42bcfed..7c447a2 100644
--- a/drivers/net/wireless/ath/ath12k/testmode.c
+++ b/drivers/net/wireless/ath/ath12k/testmode.c
@@ -155,8 +155,9 @@ static int ath12k_tm_cmd_wmi(struct ath12k *ar, struct nlattr *tb[])
 	struct ath12k_pdev_wmi *wmi = ar->wmi;
 	struct sk_buff *skb;
 	u32 cmd_id, buf_len;
-	int ret;
+	int ret, tag;
 	void *buf;
+	u32 *ptr;
 
 	mutex_lock(&ar->conf_mutex);
 
@@ -179,6 +180,13 @@ static int ath12k_tm_cmd_wmi(struct ath12k *ar, struct nlattr *tb[])
 	buf_len = nla_len(tb[ATH12K_TM_ATTR_DATA]);
 	cmd_id = nla_get_u32(tb[ATH12K_TM_ATTR_WMI_CMDID]);
 
+	ptr = (u32 *)buf;
+	tag = FIELD_GET(WMI_TLV_TAG, *ptr);
+	ptr++;
+
+	if (tag == WMI_TAG_PDEV_SET_PARAM_CMD)
+		*ptr = ar->pdev->pdev_id;
+
 	ath12k_dbg(ar->ab, ATH12K_DBG_TESTMODE,
 		   "testmode cmd wmi cmd_id %d buf %pK buf_len %d\n",
 		   cmd_id, buf, buf_len);
-- 
2.17.1

